src/lem/circuit.rs

```diff
@@ -513,10 +584,35 @@ impl Func {
                         let allocated_ptr = AllocatedPtr::from_parts(tag, src.hash().clone());
                         bound_allocations.insert(tgt.clone(), allocated_ptr);
                     }
+                    Op::EqTag(tgt, a, b) => {
+                        let a = bound_allocations.get(a)?;
+                        let b = bound_allocations.get(b)?;
+                        let a_num = a.tag();
+                        let b_num = b.tag();
+                        let eq = alloc_equal(&mut cs.namespace(|| "equal_tag"), a_num, b_num)?;
+                        let c_num = boolean_to_num(&mut cs.namespace(|| "equal_tag.to_num"), &eq)?;
```

Comment:
Does this mean that we are actually manifesting this 'abuse of Lurk tags' for internal LEM operations in the circuit itself? This seems very wrong, and if we can avoid it, I'd like to.

It looks like we're producing a 'boolean-like' value, tagged as Lurk data, *in the circuit*, even though it is neither the right way to represent booleans *or* Lurk data in the circuit.